build/npm/preinstall.js

```diff
@@ -28,7 +28,47 @@ if (!/yarn[\w-.]*\.js$|yarnpkg$/.test(process.env['npm_execpath'])) {
 	err = true;
 }
 
+const os = process.env['OS'];
+if (os.startsWith('Windows')) {
+	const vsVersion = getLatestSupportedVisualStudioVersion();
+	if (!vsVersion) {
+		console.error('\033[1;31m*** Please use Visual Studio 2017 or 2019.\033[0;0m');
```

Comment:
Developer starting with vscode repo don't need complete suite of VS. They can install the minimal requirement following our guide https://github.com/microsoft/vscode/wiki/How-to-Contribute, can you point them to the link here